### PR TITLE
DOC: contributing guide: point users to environment file to install deps

### LINKED ISSUE #2220: DOC: contributing guide: point users to env file to install deps?
#### Is your feature request related to a problem?

No

#### Describe the solution you'd like

Switch https://docs.geopandas.org/en/latest/community/contributing.html#installing-dependencies to point to a "geopandas_dev" environment file and put it in e.g. https://github.com/geopandas/geopandas/blob/master/ci/envs

Rather than have a user do a few manual import steps e.g.

```
conda create -n geopandas_dev python
conda activate geopandas_dev
conda install -c conda-forge pandas fiona shapely pyproj rtree pytest
python -m pip install pre-commit
pre-commit install
```

Create a geopandas_dev file which has https://github.com/geopandas/geopandas/blob/master/ci/envs/39-latest-conda-forge.yaml in it as well as others e.g. pre-commit and have a user do `conda/mambe env update -f 39-latest-conda-forge.yaml`

#### Additional context

https://github.com/geopandas/geopandas/blob/master/ci/envs/39-latest-conda-forge.yaml could be used but it's named "test" and doesn't have pre-commit.

### PR DESCRIPTION
**Pull request by wxman22**
Updated existing dev environment file and documentation to resolve #2220

### CODE REVIEW

File: doc/source/community/contributing.rst [hunk 1]
```diff
@@ -155,14 +155,33 @@ An easy way to create a *GeoPandas* development environment is as follows:
 - Make sure that you have :ref:`cloned the repository <contributing.forking>`
 - ``cd`` to the *geopandas** source directory
 
-Tell conda to create a new environment, named ``geopandas_dev``, or any other name you would like
+Using the provided environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*GeoPandas* provides an environment which includes the required dependencies for development.
+The environment file is located in the top level of the repo and is named ``environment-dev.yml``.
+You can create this environment by navigating to the git clone of the *GeoPandas* repository
```
Comment (martinfleis) [inline doc/source/community/contributing.rst:163]:
```suggestion
You can create this environment by navigating to the the *GeoPandas* source directory
```

That is in line with the wording we use above.

File: environment-dev.yml [hunk 2]
```diff
@@ -2,31 +2,39 @@ name: geopandas-dev
 channels:
     - conda-forge
 dependencies:
+    - python=3.9
```
Comment (martinfleis) [inline environment-dev.yml:5]:
```suggestion
    - python
```

I don't think we need a pin here. Using 3.10 is fine as well.

**Review by martinfleis — COMMENTED** [review]
Thanks! This is super helpful for new contributors. I've left a few minor comments but otherwise it looks good.

**Review by martinfleis — APPROVED** [review]
thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**